backend/tasks.js

```diff
@@ -30,56 +30,96 @@ export const TASKS = {
 export function executeTask(responseTools, responseSender, payload) {
 
 	/*
-	 *	respond with responseSender(wantedResponse) in sequelizeManager
+	 *	respond with responseSender(wantedResponse)
 	 *	payload used to indentify which task to perform and on what message
 	 */
 
     // unpack tools
-	const {sequelizeManager} = responseTools;
-
-    // shorthands
-    const log = (mes, lvl) => sequelizeManager.log(mes, lvl);
-    const raiseError = (err, res) => sequelizeManager.raiseError(err, res);
+    const {sequelizeManager, elasticManager} = responseTools;
 
     // unpack payload
 	const {
 		task,
-		sessionSelected = sequelizeManager.sessionSelected,
+		sessionSelected = null,
 		message = null // optional for some tasks
 	} = payload;
 
-	/*
+    // unpack sessionSelected and decide which responseManager to use
+    console.log(sessionSelected);
+    const sequelizeSessions = Object.keys(sequelizeManager.sessions);
+    const elasticSessions = Object.keys(elasticManager.sessions);
+    const isElasticSession = contains(sessionSelected, elasticSessions);
+    const isSequelizeSession = contains(sessionSelected, sequelizeSessions);
+    console.log(sequelizeSessions);
+    console.log(elasticSessions);
+    console.log(isElasticSession);
+    console.log(isSequelizeSession);
+
+    let responseManager;
+    if (isElasticSession && !isSequelizeSession) {
+        console.log('using elasticManager');
+        responseManager = elasticManager;
+    } else if (!isElasticSession && isSequelizeSession) {
+        console.log('using sequelize');
+        responseManager = sequelizeManager;
+    } else if (!isElasticSession && !isSequelizeSession) {
+        console.log('session does not exist');
+        // see if new elasticSearch session
+        if (message) {
+            if (message.dialect) {
+                console.log(message.dialect);
+                responseManager = message.dialect === 'elasticsearch' ? elasticManager : sequelizeManager;
+            }
```

Comment:
this check for a new `responseManger` seems pretty implicit. not sure if this is something to address or not, still wrapping my head around what's going on here